src/commands/consolidate.py

```diff
@@ -262,19 +264,20 @@ async def _validate_public_keys(
 ) -> list[tuple[HexStr, HexStr]]:
     """Validate that provided public keys can be consolidated."""
     logger.info('Checking selected validators for consolidation...')
-    active_validators = await fetch_active_validators(source_public_keys + [target_public_key])
-    active_validators_public_keys = [key.public_key for key in active_validators]
-    for key in source_public_keys + [target_public_key]:
-        if key not in active_validators_public_keys:
-            raise click.ClickException(f'Trying to consolidate non-active validator {key}.')
+    all_public_keys = source_public_keys + [target_public_key]
+    active_validators = await fetch_non_exiting_validators(source_public_keys + [target_public_key])
+    active_public_keys = {key.public_key for key in active_validators}
+    non_active_public_keys = set(all_public_keys) - active_public_keys
+    for key in non_active_public_keys:
+        raise click.ClickException(f'Trying to consolidate non-active validator {key}.')
 
     # validate that target_public_key is a compounding validator.
     # Not required for a single validator consolidation
-    if not len(source_public_keys) == 1 and source_public_keys[0] != target_public_key:
-        target_validator = [
-            val for val in active_validators if val.public_key == target_public_key
-        ][0]
-        if not target_validator.is_compounding:
+    if not _is_switch_to_compounding(source_public_keys, target_public_key):
+        target_validator = find_first(
+            active_validators, lambda val: val.public_key == target_public_key
+        )
+        if target_validator and not target_validator.is_compounding:
```

Comment:
`target_validator` is not None at this point, because of validation above. So it's better to `cast`. For example:

```suggestion
        # `target_validator` is not None, because of validation above
        target_validator = cast(ConsensusValidator, target_validator)
        if not target_validator.is_compounding:
```